libs/openFrameworks/sound/ofAVEngineSoundPlayer.m

```diff
@@ -0,0 +1,568 @@
+//
+//  ofAVEngineSoundPlayer.cpp
+//  soundPlayerExample
+//
+//  Created by Theo Watson on 3/24/21.
+//
+
+#include "ofAVEngineSoundPlayer.h"
+#include "ofUtils.h"
+#include "ofMath.h"
+#include "ofLog.h"
+
+#define TARGET_OSX
+
+//REFS: https://github.com/ooper-shlab/AVAEMixerSample-Swift/blob/master/AVAEMixerSample/AudioEngine.m
+// https://developer.apple.com/documentation/avfaudio/avaudioengine
+// https://developer.apple.com/forums/thread/14138
+// https://developer.apple.com/forums/thread/48442
+// https://github.com/garynewby/XYAudioView/blob/master/XYAudioView/BasicAVAudioEngine.m
+// https://github.com/twilio/video-quickstart-ios/blob/master/AudioDeviceExample/AudioDevices/ExampleAVAudioEngineDevice.m
+
+#import <Foundation/Foundation.h>
+#import <AVFoundation/AVFoundation.h>
+
+@interface AVEnginePlayer : NSObject
+
+@property(nonatomic, retain) NSTimer * timer;
+
+- (BOOL)loadWithFile:(NSString*)file;
+- (BOOL)loadWithPath:(NSString*)path;
+- (BOOL)loadWithURL:(NSURL*)url;
+
+- (void)unloadSound;
+
+- (void)play;
+- (void)play: (float)startTime;
+- (void)pause;
+- (void)stop;
+
+- (BOOL)isLoaded;
+- (BOOL)isPlaying;
+
+- (void)volume:(float)value;
```

Comment:
this is a small part, getter/setter method in basic obj-c style is

```
- (void)setVolume:(float)value;
- (float)volume;
```

then we can use property style access like

```
obj.volume = 0.5f; // equivalent to [obj setVolume:0.5f];
float v = obj.volume; // equivalent to float v = [obj volume];
```